tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php

```diff
@@ -2,48 +2,88 @@
+        $dummySid = $this->createMock(Sid::class);
 
         $this->mockRepository->expects($this->once())
-            ->method('findAllByServiceAndDateRange')
-            ->with($sid, $fromDateTime, $toDatetime, -1, 0)
-            ->willReturn($dbData);
-
-        $broadcasts = $this->service()->findByServiceAndDateRange(
-            $sid,
-            $fromDateTime,
-            $toDatetime,
-            -1,
-            1
-        );
+             ->method('findAllByServiceAndDateRange')
+             ->with(
+                 $dummySid,
+                 $dummyFromDate,
+                 $dummyToDate,
+                 $expectedLimit,
+                 $expectedOffset
+             );
 
-        $this->assertEquals($this->broadcastsFromDbData($dbData), $broadcasts);
+        $this->service()->findByServiceAndDateRange($dummySid, $dummyFromDate, $dummyToDate, ...$paginationParams);
     }
 
-    public function testFindByServiceAndDateRangeWhenNoBroadcastsFound()
+    public function paginationProvider(): array
     {
-        $fromDateTime = new DateTimeImmutable('2010-01-15 06:00:00');
-        $toDatetime = new DateTimeImmutable('2017-10-16 06:00:00');
-        $sid = new Sid('this_sid_doesnt_exist');
+        return [
+            // [expectedLimit, expectedOffset, [limit, page]]
+            'default pagination' => [300, 0, []],
+            'custom pagination' => [3, 12, [3, 5]],
+        ];
+    }
 
+    /**
+     * @dataProvider repositoryResultsProvider
+     */
+    public function testFindByServiceAndDateRange($expectedPids, $stubRepositoryResults)
+    {
         $this->mockRepository->expects($this->once())
             ->method('findAllByServiceAndDateRange')
-            ->with($sid, $fromDateTime, $toDatetime, -1, 0)
-            ->willReturn([]);
+            ->willReturn($stubRepositoryResults);
+
+        $stubBroadcasts = $this->service()->findByServiceAndDateRange(
+            $this->createMock(Sid::class),
+            $this->createMock(DateTimeImmutable::class),
+            $this->createMock(DateTimeImmutable::class)
+        );
 
-        $broadcasts = $this->service()
-            ->findByServiceAndDateRange($sid, $fromDateTime, $toDatetime, -1, 1);
+        $this->assertCount(count($expectedPids), $stubBroadcasts);
+        $this->assertContainsOnly(Broadcast::class, $stubBroadcasts);
+        $this->assertSame($expectedPids, $this->extractPids($stubBroadcasts));
+    }
+
+    public function repositoryResultsProvider(): array
+    {
+        return [
+            [
+                ['b00swyx1', 'b010t150'],
+                [['pid' => 'b00swyx1'], ['pid' => 'b010t150']],
+            ],
+            [
+                [],
+                [],
+            ],
+        ];
+    }
 
-        $this->assertSame([], $broadcasts);
+    /**
+     * @param Broadcast[] $broadcasts
+     * @return string[]
+     */
+    private function extractPids(array $broadcasts): array
```

Comment:
We're going to use this in pretty much all services as they follow a similar pattern right? Would it be better placed in the AbstractServiceTest so everything can use it